BangazonWorkForceManagement/Controllers/DepartmentsController.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data.SqlClient;
+using System.Linq;
+using System.Threading.Tasks;
+using BangazonAPI.Models;
+using BangazonWorkForceManagement.Models.ViewModels;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+
+namespace BangazonWorkForceManagement.Controllers
+{
+    public class DepartmentsController : Controller
+    {
+        private readonly IConfiguration _configuration;
+
+        public DepartmentsController(IConfiguration configuration)
+        {
+            this._configuration = configuration;
+        }
+
+        public SqlConnection Connection
+        {
+            get
+            {
+                return new SqlConnection(_configuration.GetConnectionString("DefaultConnection"));
+            }
+        }
+
+
+        // GET: Departments
+        public ActionResult Index()
+        {
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = @"SELECT id, [name], budget FROM Department";
+                    SqlDataReader reader = cmd.ExecuteReader();
+
+                    List<Department> departments = new List<Department>();
+
+                    while (reader.Read())
+                    {
+                        Department department = new Department
+                        {
+                            Id = reader.GetInt32(reader.GetOrdinal("id")),
+                            Name = reader.GetString(reader.GetOrdinal("name")),
+                            Budget = reader.GetInt32(reader.GetOrdinal("budget")),
+                            Employees = GetDeptEmployee(reader.GetInt32(reader.GetOrdinal("id")))
+                        };
+                        departments.Add(department);
+
+                    }
+                        reader.Close();
+                        return View(departments);
+                }
+            }
+        }
+
+        // GET: Departments/Details/5
+        public ActionResult Details(int id)
+        {
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = $@"SELECT id, [name], budget FROM Department WHERE id = {id}";
```

Comment:
Use `SqlParamter`s. Never use string interpolation for sql

Reply:
👍